docs/references/file_provider_markdown.md

```diff
@@ -0,0 +1,272 @@
+File Provider
+=============
+
+The File Provider lets you test a Python SDK tool independent of Alteryx
+Designer.
+
+Installation
+------------
+
+Perform these steps to set up your environment correctly:
+
+1.  Activate the base virtual environment.
+2.  Run `pip install /path/to/ayx-sdks/gitrepo`.
+
+The -e flag (for example, `pip install -e/path/to/ayx-sdks/gitrepo`)
+installs in editable mode so that changes made to the repo are
+automatically reflected.
+
+Input File Formats
+------------------
+
+The File Provider needs 4 types of input files to run.
+
+### Tool Configuration File
+
+The tool configuration file contains user input information. In
+Designer, this information is normally chosen through the SDK\'s UI, but
+in the File Provider these values are set in this configuration file.
+
+An empty configuration file contains only a `Configuration` tag, and a
+configuration file with user input has a tag corresponding to each value
+that a user sets. See the example below.
+
+```xml
+<Configuration>
+    <Value>10</Value>
+</Configuration>
+```
+
+### Workflow Configuration File
+
+Each kind of example tool has its own XML configuration file. The file
+contains tool information like name, description, input anchors, and
+output anchors. The input and output anchor information is found under
+the `InputConnections` and `OutputConnections` tags respectively. Each
+`Connection` corresponds to 1 anchor. If the `AllowMultiple` flag is
+`True` then there can be multiple connections attached to that anchor.
+
+### Metadata File
+
+The metadata XML file contains information about each field of the
+incoming data. Each field must have a `name`, `size`, and `type`.
+
+-   `name`: The field name.
+-   `size`: The size of each piece of data in that field.
+-   `type`: The type of each piece of data in that field.
+
+`type` must correspond to 1 of the values in the `FieldType` enum. See
+the example below:
+
+```xml
+<RecordInfo>
+  <Field name="Name" size="254" type="v_string" />
+  <Field name="Value" size="254" type="double" scale="2" description="the value" source="internet"/>
+</RecordInfo>
+```
+
+### Record File
+
+The record CSV file contains all of the record information. It looks
+like a CSV file that Designer might use as input. The first row should
+contain field names corresponding to the name of each field in the
+metadata file. The rest of the rows should contain the record data. See
+the example below:
+
+```cs
+Name,Value
+Ten,10
+Twenty,20
+Thirty,30
+```
+
+Running the File Provider
+-------------------------
+
+The file provider is run from a command-line argument. The only argument
+the user must pass in is a path to a JSON file with this format:
```

Comment:
TW review:

The file provider is run from a command-line argument. The only argument
you must pass in, is a path to a JSON file with this format:

Reply:
fixed